freqtrade/optimize/backtesting.py

```diff
@@ -691,15 +693,50 @@ def start(self) -> None:
         self.load_bt_data_detail()
         logger.info("Dataload complete. Calculating indicators")
 
+        run_ids = {
+            strategy.get_strategy_name(): get_strategy_run_id(strategy)
+            for strategy in self.strategylist
+        }
+
+        # Load previous result that will be updated incrementally.
+        if self.config.get('timerange', '-').endswith('-'):
```

Comment:
To me, this check is potentially wrong / erroneous

you should rather use `self.timerange` and evaluate if the end-date is present and not after today.

consider the following "workaround":
`--timerange 20220101-20230101` - it would pass your test here just fine (doesn't end with "-") - However - my end-date is 11 months in the future, so i'll be using incremental data "at will".